site/src/contexts/LocalPreferencesContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react"
+
+const LOCAL_PREFERENCES_KEY = "local-preferences"
+
+const defaultValues = {
+  buildLogsVisibility: "visible" as "visible" | "hide",
+}
+
+type LocalPreferencesValues = typeof defaultValues
+type LocalPreference = keyof LocalPreferencesValues
+type LocalPreferenceContextValues = {
+  values: LocalPreferencesValues
+  getPreference: (
+    name: LocalPreference,
+  ) => LocalPreferencesValues[LocalPreference]
+  setPreference: (
+    name: LocalPreference,
+    value: LocalPreferencesValues[LocalPreference],
+  ) => void
+}
+
+const LocalPreferencesContext = createContext<
+  LocalPreferenceContextValues | undefined
+>(undefined)
+
+export const LocalPreferenceProvider = ({
+  children,
+}: {
+  children: ReactNode
+}) => {
+  const [state, setState] = useState<{
+    ready: boolean
+    values: LocalPreferencesValues
+  }>({ ready: false, values: defaultValues })
+
+  useEffect(() => {
+    const preferencesStr = window.localStorage.getItem(LOCAL_PREFERENCES_KEY)
+    if (preferencesStr) {
+      try {
+        const values = JSON.parse(preferencesStr)
+        setState({ ready: true, values })
```

Comment:
Should this be something like:

```suggestion
        setState({ ready: true, values: values.values })
```

Or maybe:
```suggestion
        setState({ ...values, ready: true })
```

The values in my local storage are getting nested like this:

```json
{"ready":true,"values":{"ready":true,"values":{"ready":true,"values":{"ready":true,"values":{"buildLogsVisibility":"visible"},"buildLogsVisibility":"visible"},"buildLogsVisibility":"visible"},"buildLogsVisibility":"visible"}}
```

Reply:
Ahhh good catch * 2